src/uwtools/drivers/driver.py

```diff
@@ -109,8 +119,12 @@ def _driver_config(self) -> Dict[str, Any]:
         """
         Returns the config block specific to this driver.
         """
-        driver_config: Dict[str, Any] = self._config[self._driver_name]
-        return driver_config
+        name = self._driver_name
+        try:
+            driver_config: Dict[str, Any] = self._config[name]
+            return driver_config
+        except KeyError as e:
+            raise UWConfigError("Required '%s' block missing in config" % name) from e
```

Comment:
Old behavior when e.g. the `ungrib` block was missing from the YAML config:
```
% uw ungrib provisioned_run_directory --cycle 2024-03-22T18 --config ungrib.yaml --batch
[2024-03-22T23:55:23]     INFO Validating config against internal schema ungrib
[2024-03-22T23:55:23]     INFO 0 UW schema-validation errors found
[2024-03-22T23:55:23]     INFO Validating config against internal schema platform
[2024-03-22T23:55:23]     INFO 0 UW schema-validation errors found
[2024-03-22T23:55:23]     INFO 20240322 18Z ungrib provisioned run directory: Initial state: Pending
[2024-03-22T23:55:23]     INFO 20240322 18Z ungrib provisioned run directory: Checking requirements
[2024-03-22T23:55:23]     INFO 20240322 18Z ungrib /tmp/20240322/18/run/GRIBFILE.AAA: Initial state: Pending
[2024-03-22T23:55:23]     INFO 20240322 18Z ungrib /tmp/20240322/18/run/GRIBFILE.AAA: Checking requirements
[2024-03-22T23:55:23]  WARNING File /path/to/dir/gfs.t12z.pgrb2.0p25.f000: State: Pending (EXTERNAL)
[2024-03-22T23:55:23]     INFO 20240322 18Z ungrib /tmp/20240322/18/run/GRIBFILE.AAA: Requirement(s) pending
[2024-03-22T23:55:23]  WARNING 20240322 18Z ungrib /tmp/20240322/18/run/GRIBFILE.AAA: Final state: Pending
[2024-03-22T23:55:23]     INFO 20240322 18Z ungrib /tmp/20240322/18/run/namelist.wps: Initial state: Pending
[2024-03-22T23:55:23]     INFO 20240322 18Z ungrib /tmp/20240322/18/run/namelist.wps: Checking requirements
[2024-03-22T23:55:23]     INFO 20240322 18Z ungrib /tmp/20240322/18/run/namelist.wps: Requirement(s) ready
[2024-03-22T23:55:23]     INFO 20240322 18Z ungrib /tmp/20240322/18/run/namelist.wps: Executing
[2024-03-22T23:55:23]     INFO 20240322 18Z ungrib /tmp/20240322/18/run/namelist.wps: Final state: Ready
[2024-03-22T23:55:23]     INFO 20240322 18Z ungrib runscript.ungrib: Initial state: Pending
[2024-03-22T23:55:23]     INFO 20240322 18Z ungrib runscript.ungrib: Checking requirements
[2024-03-22T23:55:23]     INFO 20240322 18Z ungrib runscript.ungrib: Requirement(s) ready
[2024-03-22T23:55:23]     INFO 20240322 18Z ungrib runscript.ungrib: Executing
Traceback (most recent call last):
  File "/home/maddenp/sw/conda/envs/DEV-uwtools/bin/uw", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/home/maddenp/git/uwtools/src/uwtools/cli.py", line 68, in main
    sys.exit(0 if modes[args[STR.mode]](args) else 1)
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/maddenp/git/uwtools/src/uwtools/cli.py", line 707, in _dispatch_ungrib
    return uwtools.api.ungrib.execute(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/maddenp/git/uwtools/src/uwtools/api/ungrib.py", line 35, in execute
    getattr(obj, task)()
  File "/home/maddenp/sw/conda/envs/DEV-uwtools/lib/python3.11/site-packages/iotaa/__init__.py", line 453, in __iotaa_tasks__
    assets = _delegate(g, taskname)
             ^^^^^^^^^^^^^^^^^^^^^^
  File "/home/maddenp/sw/conda/envs/DEV-uwtools/lib/python3.11/site-packages/iotaa/__init__.py", line 480, in _delegate
    alist = list(filter(None, chain(*[_listify(a) for a in _listify(next(g))])))
                                                                    ^^^^^^^
  File "/home/maddenp/git/uwtools/src/uwtools/drivers/ungrib.py", line 97, in provisioned_run_directory
    self.runscript(),
    ^^^^^^^^^^^^^^^^
  File "/home/maddenp/sw/conda/envs/DEV-uwtools/lib/python3.11/site-packages/iotaa/__init__.py", line 428, in __iotaa_task__
    _execute(g, taskname)
  File "/home/maddenp/sw/conda/envs/DEV-uwtools/lib/python3.11/site-packages/iotaa/__init__.py", line 497, in _execute
    next(g)
  File "/home/maddenp/git/uwtools/src/uwtools/drivers/ungrib.py", line 110, in runscript
    self._write_runscript(path=path, envvars={})
  File "/home/maddenp/git/uwtools/src/uwtools/drivers/driver.py", line 238, in _write_runscript
    scheduler=self._scheduler if self._batch else None,
              ^^^^^^^^^^^^^^^
  File "/home/maddenp/git/uwtools/src/uwtools/drivers/driver.py", line 220, in _scheduler
    return JobScheduler.get_scheduler(self._resources)
                                      ^^^^^^^^^^^^^^^
  File "/home/maddenp/git/uwtools/src/uwtools/drivers/driver.py", line 145, in _resources
    "account": self._config["platform"]["account"],
               ~~~~~~~~~~~~^^^^^^^^^^^^
  File "/home/maddenp/sw/conda/envs/DEV-uwtools/lib/python3.11/collections/__init__.py", line 1126, in __getitem__
    raise KeyError(key)
KeyError: 'platform'
```
New behavior:
```
% uw ungrib provisioned_run_directory --cycle 2024-03-22T18 --config ungrib.yaml --batch
[2024-03-23T00:25:45]     INFO Validating config against internal schema ungrib
[2024-03-23T00:25:45]     INFO 0 UW schema-validation errors found
[2024-03-23T00:25:45]     INFO Validating config against internal schema platform
[2024-03-23T00:25:45]     INFO 0 UW schema-validation errors found
[2024-03-23T00:25:45]     INFO 20240322 18Z ungrib provisioned run directory: Initial state: Pending
[2024-03-23T00:25:45]     INFO 20240322 18Z ungrib provisioned run directory: Checking requirements
[2024-03-23T00:25:45]     INFO 20240322 18Z ungrib /tmp/20240322/18/run/GRIBFILE.AAA: Initial state: Pending
[2024-03-23T00:25:45]     INFO 20240322 18Z ungrib /tmp/20240322/18/run/GRIBFILE.AAA: Checking requirements
[2024-03-23T00:25:45]  WARNING File /path/to/dir/gfs.t12z.pgrb2.0p25.f000: State: Pending (EXTERNAL)
[2024-03-23T00:25:45]     INFO 20240322 18Z ungrib /tmp/20240322/18/run/GRIBFILE.AAA: Requirement(s) pending
[2024-03-23T00:25:45]  WARNING 20240322 18Z ungrib /tmp/20240322/18/run/GRIBFILE.AAA: Final state: Pending
[2024-03-23T00:25:45]     INFO 20240322 18Z ungrib runscript.ungrib: Initial state: Pending
[2024-03-23T00:25:45]     INFO 20240322 18Z ungrib runscript.ungrib: Checking requirements
[2024-03-23T00:25:45]     INFO 20240322 18Z ungrib runscript.ungrib: Requirement(s) ready
[2024-03-23T00:25:45]     INFO 20240322 18Z ungrib runscript.ungrib: Executing
[2024-03-23T00:25:45]    ERROR Required 'platform' block missing in config
```
Similar handling is provided for the `platform` block, below.